app/models/admin/collection.rb

```diff
@@ -271,6 +281,14 @@ def create_dropbox_directory!
       end
     end
 
+    def destroy_dropbox_directory!
+      if Settings.dropbox.path =~ %r(^s3://)
+        destroy_s3_dropbox_directory!
+      else
+        destroy_fs_dropbox_directory!
+      end
```

Comment:
IMO these (and the other) directory handling logic would be better off in a service class (FileLocator or something like that)